apps/src/assetManagement/animationLibraryApi.js

```diff
@@ -90,6 +99,113 @@ export function createDefaultSpriteMetadata(listData) {
   });
 }
 
+export function buildAnimationMetadata(files) {
+  let animationMetadataByName = {};
+  let resolvedPromisesArray = [];
+  for (const [fileKey, fileObject] of Object.entries(files)) {
+    let json = fileObject['json'];
+    let png = fileObject['png'];
+    resolvedPromisesArray.push(
+      getAnimationLibraryFile(json.key)
+        .then(metadata => {
+          // Metadata contains name, frameCount, frameSize, looping, frameDelay
+          let combinedMetadata = metadata;
+          combinedMetadata['jsonLastModified'] = json.last_modified;
+          combinedMetadata['pngLastModified'] = png.last_modified;
+          combinedMetadata['version'] = png.version_id;
+          combinedMetadata[
+            'sourceUrl'
+          ] = `/api/v1/animation-library/level_animations/${png.version_id}/${
+            metadata.name
+          }.png`;
+          combinedMetadata['sourceSize'] = png.source_size;
+          animationMetadataByName[fileKey] = combinedMetadata;
+          return Promise.resolve();
+        })
+        .catch(err => {
+          return Promise.reject(err);
+        })
+    );
+  }
+  return Promise.all(resolvedPromisesArray).then(() => {
+    return animationMetadataByName;
+  });
+}
+
+export function buildAliasMap(animationMetadata) {
+  let aliasMap = {};
+
+  for (const [key, metadata] of Object.entries(animationMetadata)) {
+    let aliases = [metadata['name'].toLowerCase()];
+    if (metadata['aliases']) {
+      metadata['aliases'].map(alias => {
+        aliases.push(alias.toLowerCase());
+      });
+    }
+    aliases.map(alias => {
+      //Push name into target array, deduplicate, and sort
+      let updatedMap = {...aliasMap};
+      let addedSet = new Set(updatedMap[alias]);
+      addedSet.add(key);
+      aliasMap[alias] = [...addedSet].sort();
+    });
+  }
+  return aliasMap;
+}
+
+export function buildCategoryMap(animationMetadata) {
+  let categoryMap = {};
+
+  for (const [key, metadata] of Object.entries(animationMetadata)) {
+    let categories = metadata['categories'];
+    // If the animation doesn't have a category, place it in a section for
+    // level-specific/hidden-from-library animations.
+    if (!categories) {
+      categories = ['level_animations'];
```

Comment:
do we use `'level_animations'` elsewhere? should this be a constant?

Reply:
We use it here and in a url. It feels just shy of making a constant, but it we end up using it more, I think it would make sense to use as a constant.